src/restLevelClient/generateParameterTypes.ts

```diff
@@ -82,6 +108,67 @@ export function generateParameterInterfaces(
   ]);
 }
 
+function getRequestHeaderInterfaceDefinition(
+  parameters: Parameter[],
+  baseName: string
+): undefined | InterfaceDeclarationStructure {
+  // Check if there are any required headers
+  const headerParameters = parameters.filter(p => p.protocol.http?.in === "header");
+  if (!headerParameters.length) {
+    return undefined;
+  }
+  const headersInterfaceName = `${baseName}Headers`;
+  return {
+    kind: StructureKind.Interface,
+    isExported: true,
+    name: headersInterfaceName,
+    properties: headerParameters.map((h: Parameter) => {
+      const description = getLanguageMetadata(h.language).description;
+      return {
+        name: `"${getLanguageMetadata(h.language).name}"`,
```

Comment:
Do headers have a serializaedName? If they do, we'll need to use that one instead of `name` name can be changed by M4 if `x-ms-client-name` is set in the swagger. `serializedName` would be the value that the service understands so that'd be the one we want to use

Reply:
I have changed it into serializedName now